src/charm.py

```diff
@@ -200,9 +201,7 @@ def change_config(self, charm_state: CharmState) -> None:
             return
         self.model.unit.status = ops.MaintenanceStatus("Configuring Synapse")
         try:
-            pebble.change_config(
-                charm_state, container, is_main=self.is_main(), unit_number=self.get_unit_number()
-            )
+            pebble.change_config(charm_state, container, unit_number=self.get_unit_number())
```

Comment:
This function, `change_config`, calls `self.main_unit`, so at this point the `charm_state.main_unit` can be outdated. I believe this can be a source of confusion (if not a bug somewhere).

Reply:
I'm not finding the self.main_unit call in change_config, where is it?